reproducibility_project/project-analysis.py

```diff
@@ -171,455 +238,305 @@ def nvt_rdf_calculated(job):
+    create_largest_t0_operations(
+        ensemble=ensemble,
+        prop_list=prop_list,
+        simulation_type="md",
+        engine_list=md_engines,
     )
 
-
-@Project.operation
-@Project.pre(lambda job: job.isfile("log-nvt.txt"))
-@Project.post(
-    lambda job: job.doc.get("nvt/sampling_results", {}).get("potential_energy")
-)
-@flow.with_job
-def determine_nvt_potential_energy_sampling(job):
-    """Write out sampling results for NVT production properties."""
-    from reproducibility_project.src.analysis.sampler import sample_job
-
-    _determine_sampling_information(
-        job=job, ensemble="nvt", prop="potential_energy", filename=None
+# mc operations
+for ensemble, prop_list in zip(["npt", "nvt"], [mc_npt_props, mc_nvt_props]):
+    create_largest_t0_operations(
+        ensemble=ensemble,
+        prop_list=prop_list,
+        simulation_type="mc",
+        engine_list=mc_engines,
     )
 
 
-@Project.operation
-@Project.pre(lambda job: job.isfile("log-nvt.txt"))
-@Project.post(
-    lambda job: job.doc.get("nvt/sampling_results", {}).get("kinetic_energy")
-)
-@flow.with_job
-def determine_nvt_kinetic_energy_sampling(job):
-    """Write out sampling results for NVT production properties."""
-    from reproducibility_project.src.analysis.sampler import sample_job
+def create_write_subsampled_max_t0(
+    ensemble: str,
+    prop_list: List[str],
+    simulation_type: str,
+    engine_list: List[str],
+    is_monte_carlo: bool,
+):
+    """Dynamically create the operations to write the subsampled data based on max t0."""
 
-    _determine_sampling_information(
-        job=job, ensemble="nvt", prop="kinetic_energy", filename=None
+    @Project.operation(
+        f"{ensemble}_{simulation_type}_write_subsampled_data_max_t0"
     )
-
-
-@Project.operation
-@Project.pre(lambda job: job.isfile("log-nvt.txt"))
-@Project.post(lambda job: job.doc.get("nvt/sampling_results", {}).get("volume"))
-@flow.with_job
-def determine_nvt_volume_sampling(job):
-    """Write out sampling results for NVT production properties."""
-    from reproducibility_project.src.analysis.sampler import sample_job
-
-    _determine_sampling_information(
-        job=job, ensemble="nvt", prop="volume", filename=None
+    @Project.pre(lambda job: job.isfile(f"log-{ensemble}.txt"))
+    @Project.pre(lambda job: job.doc.get(f"{ensemble}/max_t0") is not None)
+    @Project.pre(lambda job: job.sp.engine in engine_list)
+    @Project.post(
+        lambda job: all(
+            [job.isfile(f"{ensemble}_{prop}.h5") for prop in prop_list]
+        )
     )
+    @flow.with_job
+    def write_subsampled_data_max_t0(job):
+        """Write subsampled properties to job.stores."""
+        from reproducibility_project.src.analysis.sampler import (
+            get_decorr_samples_using_max_t0,
+        )
 
+        for prop in prop_list:
+            with job.stores[f"{ensemble}_{prop}"] as data:
+                data["property"] = get_decorr_samples_using_max_t0(
```

Comment:
I think my error might be related to this line. Is property supposed to be a variable?
```suggestion
                data[f"{property}"] = get_decorr_samples_using_max_t0(
```